code/modules/admin/fun_verbs.dm

```diff
@@ -977,40 +977,26 @@
 	if(!check_rights(R_FUN|R_MENTOR))
 		return
 
-	if(istype(usr, /mob/camera/imaginary_friend))
-		var/mob/camera/imaginary_friend/IF = usr
-		IF.deactivate()
-		return
+	var/client/C = usr.client
 
-	if(is_mentor(usr.client) && !isobserver(usr))
-		to_chat(usr, "<span class='warning'>Can only become an imaginary friend while observing.</span>")
+	if(istype(C.mob, /mob/camera/imaginary_friend))
+		var/mob/camera/imaginary_friend/IF = C.mob
+		IF.ghostize()
 		return
 
-	if(!isobserver(usr))
-		usr.client.holder.admin_ghost()
-
-	var/mob/living/L
-	switch(input("Select by:", "Imaginary Friend") as null|anything in list("Key", "Mob"))
-		if("Key")
-			var/client/C = input("Please, select a key.", "Imaginary Friend") as null|anything in sortKey(GLOB.clients)
-			if(!C)
-				return
-			L = C.mob
-		if("Mob")
-			var/mob/M = input("Please, select a mob.", "Imaginary Friend") as null|anything in sortNames(GLOB.mob_living_list)
-			if(!M)
-				return
-			L = M
-
-	if(!isobserver(usr))
+	if(is_mentor(C) && !isobserver(C.mob))
```

Comment:
```
if(!isobserver(C.mob))
	if(is_mentor(C)
		usr.client.holder.admin_ghost()
		to_chat(C, "<span class='warning'>Can only become an imaginary friend while observing.</span>")
		return
	C.holder.admin_ghost()
```

Reply:
That is changing the functionality to something completely different

Reply:
Unless I'm missing something the behavior should remain exactly the same this way.

Reply:
I want to return early for mentors, not admin ghost them, i want to only admnghost admins.

Reply:
That's what will happen with my code. Non-observer mentors will trigger `if(!isobserver(C.mob))` and `if(is_mentor(C))`, reaching the `return` before the `C.holder.admin_ghost()`.

Reply:
Though it would be more maintainable if it was something like `!is_admin` instead.

Reply:
```
	if(is_mentor(C))
		usr.client.holder.admin_ghost()
```
No it wont

Reply:
Heck, I only saw that now. Very much unintended. Remove that line and the rest follows.